cake/build-settings.cake

```diff
@@ -1,23 +1,354 @@
-#load ./ci.cake
-#load ./packaging.cake
-#load ./package-checks.cake
-#load ./test-results.cake
-#load ./package-tests.cake
-#load ./package-tester.cake
-#load ./header-check.cake
-#load ./local-tasks.cake
-
-public class BuildSettings
+public static class BuildSettings
 {
-    public BuildSettings(ISetupContext context)
-    {
+	private static BuildSystem _buildSystem;
+
+	public static void Initialize(
+      // Required parameters
+        ICakeContext context,
+        string title,
+        string githubRepository,
+
+      // Optional Parameters
+        bool suppressHeaderCheck = false,
+        string[] standardHeader = null,
+        string[] exemptFiles = null,
+        
+        string solutionFile = null,
+        string[] validConfigurations = null,
+        string githubOwner = "NUnit",
+
+		bool msbuildAllowPreviewVersion = false,
+		Verbosity msbuildVerbosity = Verbosity.Minimal,
+
+        string unitTests = null, // Defaults to "**/*.tests.dll|**/*.tests.exe" (case insensitive)
+        UnitTestRunner unitTestRunner = null, // If not set, NUnitLite is used
+        string unitTestArguments = null
+        )
+	{
+        // Required arguments
         if (context == null)
-            throw new System.ArgumentNullException(nameof(context));
+            throw new ArgumentNullException(nameof(context));
+		if (title == null)
```

Comment:
![image](https://github.com/nunit/nunit-console/assets/1055021/6e13c84b-8a59-40b1-8bc8-f0b22687ec89)

Mixed tabs and spaces here is making this hard to review. :D

Reply:
I see that. It does look OK in VS, where the tabs are set to 4, but here they are set to 8 and I don't see any way to change it. I fixed those lines, however.